src/tuner/nccl_ofi_model.c

```diff
@@ -17,15 +17,63 @@ float nccl_ofi_tuner_compute_base_cost(ncclFunc_t func, int algo, int proto)
 	return nccl_base_lat[algo][proto];
 }
 
-float nccl_ofi_tuner_compute_cost(struct nccl_ofi_tuner_model_params *params, struct nccl_ofi_tuner_model_dims *dims,
-				  const nccl_ofi_tuner_cost_table_t *base_costs,
-                                  ncclFunc_t func, int algo, int proto, int pipe_ops, size_t size)
+static float get_base_latency(const nccl_ofi_tuner_cost_table_t* base_costs, const ncclFunc_t func, const int algo,
+	const int proto)
+{
+	return (*base_costs)[func][algo][proto];
+}
+
+static float get_network_latency(struct nccl_ofi_tuner_model_params const* params, const int proto, float completion_overhead)
+{
+	return proto == NCCL_PROTO_SIMPLE
+		       ? params->net_lat + completion_overhead
+		       : params->net_lat;
+}
+
+static float get_nvlink_latency(const int algo, const int proto)
+{
+	return nccl_nvlink_lat[algo][proto];
+}
+
+float nccl_ofi_tuner_scale_bandwidth(const float bandwidth, const int proto)
+{
+	/* 8B total with 4B data and 4B flags, so take a 50% hit */
+	if (proto == NCCL_PROTO_LL) return bandwidth * 0.5;
+
+	/* 120B data and 8B flags */
+	if (proto == NCCL_PROTO_LL128) return bandwidth * 0.9375;
+
+	/* No overhead otherwise */
+	return bandwidth;
+}
+
+float nccl_ofi_tuner_compute_bandwidth(struct nccl_ofi_tuner_model_params const* params,
+                                       const ncclFunc_t func, const int algo)
+{
+	switch(func) {
+	case ncclFuncAllReduce:
+		switch(algo) {
+		case NCCL_ALGO_RING:
+			return params->internode_bw * params->num_rails * ofi_nccl_tuner_num_channels() / 2.0;
+		case NCCL_ALGO_NVLS_TREE:
+			return NCCL_OFI_MIN(params->intranode_bw, (params->internode_bw * params->num_rails) / 2.0) * ofi_nccl_tuner_num_channels();
+		case NCCL_ALGO_TREE:
+			return params->internode_bw * params->num_rails * ofi_nccl_tuner_num_channels() / 2.0;
```

Comment:
Did we not decide to combine the rails and internode_bw definitions as well?

Reply:
I want to review the bigger changes separately, this is largely here to just move code around and introduce the csv dumper.